### PR TITLE
use bladnn::conv2d for disc

### PR DESCRIPTION
**Pull request by minminsun**

### CODE REVIEW

File: tao_compiler/mlir/disc/transforms/conv_rewriter.cc [hunk 1]
```diff
@@ -267,8 +268,10 @@ struct ConvToDynamicConvConvert : public OpRewritePattern<mhlo::ConvolutionOp> {
 // having expected format.
 struct DiscConvRewriterPass
     : public ConvRewriterPassBase<DiscConvRewriterPass> {
-  explicit DiscConvRewriterPass()
-      : ConvRewriterPassBase<DiscConvRewriterPass>::ConvRewriterPassBase() {}
+  explicit DiscConvRewriterPass(int cc_major)
```
Comment (wyzero) [inline tao_compiler/mlir/disc/transforms/conv_rewriter.cc:271]: nit: better to add cc_minor as well?

File: tao_compiler/mlir/disc/transforms/conv_rewriter.cc [hunk 2]
```diff
@@ -168,7 +168,8 @@ LogicalResult inferExpectedLayout(ConvParams& params) {
   outputLayout.resize(rank);
 
   if (onGpu) {
-    if (inputTy.getElementType().isF16() && filterTy.getElementType().isF16()) {
+    if (cc_major >= 8 ||
```
Comment (wyzero) [inline tao_compiler/mlir/disc/transforms/conv_rewriter.cc:171]: what if TF32 is disabled on device with `cc_major >= 8`

Comment (qiuxiafei) [reply tao_compiler/mlir/disc/transforms/conv_rewriter.cc:171]: will this affect `-multi-cc` ?

File: tao_compiler/mlir/xla/ral/context/stream_executor_based_impl.cc [hunk 3]
```diff
@@ -1288,6 +1288,19 @@ bool PickBestAlgorithm(CudnnConvParams& params,
   return true;
 }  // namespace gpu_conv_impl
 
+static bool layout_match(const std::vector<int32_t>& ref,
+                         MemRefType<int32_t, 1> metadata) {
```
Comment (qiuxiafei) [inline tao_compiler/mlir/xla/ral/context/stream_executor_based_impl.cc:1292]: nit: better to pass `metadata` by const ref, too? It has at least 5 64-bit member inside.

File: tao_compiler/mlir/disc/transforms/conv_rewriter.cc (hunk 2 above)
Comment (minminsun) [reply tao_compiler/mlir/disc/transforms/conv_rewriter.cc]: > what if TF32 is disabled on device with cc_major >= 8

Ok, I added a check for NVIDIA_TF32_OVERRIDE.

> will this affect -multi-cc ?

In multi-cc mode, disc compiler tries to get optimal performance for the current architecture it executes on. For other architectures, functional correctness is guaranteed.

File: tao_compiler/mlir/disc/transforms/conv_rewriter.cc (hunk 1 above)
Comment (minminsun) [reply tao_compiler/mlir/disc/transforms/conv_rewriter.cc]: ok, cc_minor is added for future use.

File: tao_compiler/mlir/xla/ral/context/stream_executor_based_impl.cc (hunk 3 above)
Comment (minminsun) [reply tao_compiler/mlir/xla/ral/context/stream_executor_based_impl.cc]: Done, thanks.

**Review by qiuxiafei — APPROVED** [review]

**Pull request merged**